packages/components/src/components/infinite-data-list/infinite-data-list.tsx

```diff
@@ -9,16 +9,22 @@ type TInfiniteDatalist = Pick<
     'getRowSize' | 'onRowsRendered' | 'onScroll' | 'overscanRowCount' | 'rowRenderer'
 > & {
     className: string;
+    clearScroll?: () => void;
     data_list_className: string;
     has_more_items_to_load: boolean;
     has_filler: boolean;
     items: React.ComponentProps<typeof DataList>['data_source'];
     keyMapperFn: React.ComponentProps<typeof DataList>['keyMapper'];
     loadMoreRowsFn: <T>(params: IndexRange) => Promise<T>;
+    retain_scroll_position?: boolean;
+    scroll_to_index?: number;
 };
 
 const InfiniteDataList = ({
```

Comment:
let's have this component in p2p package since it's only being used there

Reply:
as discussed, will be taken up in a separate card.